Mathlib/NumberTheory/FLT/Three.lean

```diff
@@ -3,38 +3,122 @@ Copyright (c) 2024 Riccardo Brasca. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Riccardo Brasca
 -/
+import Mathlib.NumberTheory.FLT.Basic
 import Mathlib.Data.ZMod.Basic
+import Mathlib.RingTheory.Int.Basic
 
 /-!
 # Fermat Last Theorem in the case `n = 3`
 The goal of this file is to prove Fermat Last theorem in the case `n = 3`.
 
 ## Main results
-* `fermatLastTheoremThree_case_1`: the first case of Fermat Last Theorem when `n = 3`:
+* `fermatLastTheoremThree_case1`: the first case of Fermat Last Theorem when `n = 3`:
   if `a b c : ℕ` are such that `¬ 3 ∣ a * b * c`, then `a ^ 3 + b ^ 3 ≠ c ^ 3`.
 
 ## TODO
 Prove case 2.
 -/
 
+section case1
+
 open ZMod
 
 private lemma cube_of_castHom_ne_zero {n : ZMod 9} :
     castHom (show 3 ∣ 9 by norm_num) (ZMod 3) n ≠ 0 → n ^ 3 = 1 ∨ n ^ 3 = 8 := by
   fin_cases n <;> decide
```

Comment:
```suggestion
  revert n; decide
```
is faster (`fin_cases` tends to be slow). In 3, we had `dec_trivial!`, which would do the `revert` automatically...